src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -22,11 +22,12 @@ namespace System.Text.Encodings.Web
     public abstract class TextEncoder
     {
         // Fast cache for Ascii
-#pragma warning disable CA1825 // should not be Array.Empty<byte> as this is used as a singleton for comparisons
-        private static readonly byte[] s_noEscape = new byte[] { };
-#pragma warning restore CA1825
         private byte[][] _asciiEscape = new byte[0x80][];
-        
+
+        // Keep a reference to Array.Empty<byte> as this is used as a singleton for comparisons
+        // and there is no guarantee that Array.Empty<byte>() will always be the same instance.
+        private static readonly byte[] s_noEscape = Array.Empty<byte>();
```

Comment:
We could have before, we just needed to keep a global reference to it. If we don't keep a global reference to it, then it is possible every call to Array.Empty returns a different instance which would cause reference equality to work incorrectly.